### ISSUE TITLE
Questions about Stage 1 Training Parameters for Other Four Classes

### ISSUE THREAD

**Issue by yifeipet** [kechunl/AdaCode#7]
Hi Kechun,

Your code and work are amazing. I have questions regarding Stage 1. You have shared with us the parameter settings for portraits in Stage 1. Do other classes share the same training parameters as portrait?

If you could provide other YAML files for different classes in Stage 1, that would be very helpful. Thank you!

Sincerely,
Yifei

**Comment by kechunl** [comment]
For stage 1 pretraining, the hyperparameters for different classes are the same except changing the corresponding dataset path, number of iterations (based on the size of the dataset), and the codebook size (256 or 512 codes).

**Comment by yifeipet** [comment]
> For stage 1 pretraining, the hyperparameters for different classes are the same except changing the corresponding dataset path, number of iterations (based on the size of the dataset), and the codebook size (256 or 512 codes).

Thank you for your quick response! I have a few questions regarding datasets for stage 1:
1. I used the pretrained segformer_mit-b5_8xb2-160k_ade20k-512x512 model under the mmsegmentation github. There are 150 classes predefined in [https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/ade.py](url). Then, I classified these images based on the maximum count of predictions using MMSegInferencer. During the inference, I grouped the 150 classes into 4 groups, except for portraits, based on the paper. Have I missed anything? Do I need to mask other objects that do not belong to the class for each image for the training? Did you use the ADE20K validation dataset to do the Table 1 in the supplementary material of the paper?  If not, which datasets were used for that table?

2. Did you use the FFHQ aligned face dataset to train the codebook for portraits? 

Thank you!

Sincerely,
Yifei

**Comment by kechunl** [comment]
Hi Yifei,

For the training of the 4 subdatasets, I didn't mask the objects that don't belong to the class. For the table 1 in supp, I think we used OST dataset for evaluation.
For the ffhq dataset, yes we used the aligned dataset and sampled 10,000 images to train.

**Comment by yifeipet** [comment]
> Hi Yifei,
> 
> For the training of the 4 subdatasets, I didn't mask the objects that don't belong to the class. For the table 1 in supp, I think we used OST dataset for evaluation. For the ffhq dataset, yes we used the aligned dataset and sampled 10,000 images to train.

Thank you very much! Wishing you a Happy Spring Festival!

Sincerely,
Yifei
